masu/processor/report_processor.py

```diff
@@ -0,0 +1,480 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Processor for Cost Usage Reports."""
+
+import csv
+import gzip
+import io
+import json
+import logging
+from itertools import islice
+from os import path
+
+from masu.config import Config
+from masu.database.report_db_accessor import ReportDBAccessor
+from masu.database.reporting_common_db_accessor import ReportingCommonDBAccessor
+from masu.processor.exceptions import MasuProcessingError
+
+LOG = logging.getLogger(__name__)
+
+ALLOWED_COMPRESSIONS = ('', 'gzip')
+
+
+# pylint: disable=too-few-public-methods
+class ProcessedReport:
+    """Cost usage report transcribed to our database models.
+
+    Effectively a struct for associated database tables.
+    """
+
+    def __init__(self):
```

Comment:
Check out https://books.google.com/books?id=Q0s6Vgb98CQC&lpg=PT212&dq=Python+Cookbook+%22Collecting+a+Bunch+of+Named+Items%22&hl=en&pg=PT213#v=onepage&q&f=false

I know it offends your OO sensibilities, but Python really does blur a lot of object oriented lines. Everything is an object, nothing is truly private, duck typing, data encapsulation feels like an afterthought, et cetera. I would argue that getters and setters are not "Pythonic", and in this case they wouldn't be adding anything since I have no checks or modifications to the incoming data. A public attribute that I add to in the processor is all I need. This could be a dictionary of dictionaries and lists that I then just wipe out every time we save to the DB, but I find the attribute syntax on a class more readable.